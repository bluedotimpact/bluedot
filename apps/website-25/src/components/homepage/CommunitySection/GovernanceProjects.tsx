import { Card, CTALinkOrButton, SlideList } from '@bluedot/ui';

interface Project {
  id: number;
  title: string;
  authorName: string;
  imageSrc: string;
  ctaUrl: string;
}

const projects: Project[] = [
  {
    id: 1,
    title: 'Contextual Constitutional AI',
    authorName: 'Akshat Naik',
    imageSrc: '/images/project-winners/akshat-naik.png',
    ctaUrl: 'https://aisafetyfundamentals.com/projects/contextual-constitutional-ai/',
  },
  {
    id: 2,
    title: 'Safety Haven: Justifying and exploring an antitrust safe haven for AI safety research collaboration',
    authorName: 'Ella Duus',
    imageSrc: '/images/project-winners/ella-duus.png',
    ctaUrl: 'https://aisafetyfundamentals.com/projects/safety-haven-justifying-and-exploring-an-antitrust-safe-haven-for-ai-safety-research-collaboration/',
  },
  {
    id: 3,
    title: 'Societal Adaptation to AI Human-Labor Automation',
    authorName: 'Yuval Rymon',
    imageSrc: '/images/project-winners/yuval-rymon.png',
    ctaUrl: 'https://aisafetyfundamentals.com/projects/societal-adaptation-to-ai-human-labor-automation/',
  },
  {
    id: 4,
    title: 'Are you secretly training AI? Methods for uncovering covert AI training: A framework for feasibility and future research',
    authorName: 'Naci Cankaya',
    imageSrc: '/images/project-winners/naci-cankaya.png',
    ctaUrl: 'https://aisafetyfundamentals.com/projects/are-you-secretly-training-ai-methods-for-uncovering-covert-ai-training-a-framework-for-feasibility-and-future-research/',
  },
];

const GovernanceProjects = () => {
  return (
    <div className="governance-projects p-6 container-lined flex flex-col gap-spacing-y">
      <SlideList
        title="AI Governance Projects"
        titleLevel="h3"
        subtitle="Competition winners"
        maxItemsPerSlide={4}
        minItemWidth={260}
      >
        {projects.map((project) => (
          <Card
            key={project.title}
            title={project.title}
            subtitle={`by ${project.authorName}`}
            subtitleClassName="text-base"
            imageSrc={project.imageSrc}
            ctaUrl={project.ctaUrl}
            className="governance-projects__project"
            isEntireCardClickable
          />
        ))}
      </SlideList>
      <CTALinkOrButton
        className="governance-projects__link h-fit py-3"
        url="https://aisafetyfundamentals.com/projects/"
        variant="secondary"
        withChevron
      >
        Explore more projects
      </CTALinkOrButton>
    </div>
  );
};

export default GovernanceProjects;
