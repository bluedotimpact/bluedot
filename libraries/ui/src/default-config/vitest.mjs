/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
