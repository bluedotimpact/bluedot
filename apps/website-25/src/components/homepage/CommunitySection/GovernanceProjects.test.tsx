import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import GovernanceProjects from './GovernanceProjects';

describe('GovernanceProjects', () => {
  test('renders as expected', () => {
    const { container } = render(<GovernanceProjects />);
    expect(container).toMatchSnapshot();
  });

  test('displays correct number of projects', () => {
    const { container } = render(<GovernanceProjects />);
    const testimonialCards = container.querySelectorAll('.governance-projects__project');
    expect(testimonialCards).toHaveLength(4);
  });
});
