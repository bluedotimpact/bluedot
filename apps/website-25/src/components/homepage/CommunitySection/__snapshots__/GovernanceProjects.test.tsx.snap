// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GovernanceProjects > renders as expected 1`] = `
<div>
  <section
    class="governance-projects p-6 container-lined"
  >
    <div
      class="governance-projects__container mx-auto max-w-[1750px] flex flex-col gap-12"
    >
      <div
        class="governance-projects__title-container flex flex-col gap-2"
      >
        <h3
          class="governance-projects__title"
        >
          AI Governance Projects
        </h3>
        <p
          class="governance-projects__featured-label text-xs font-[650] uppercase text-bluedot-black"
        >
          Competition winners
        </p>
      </div>
      <div
        class="governance-projects__grid grid grid-cols-[repeat(auto-fit,minmax(300px,max-content))] gap-4 overflow-visible mb-3"
      >
        <a
          class="card flex flex-col items-start transition-transform duration-200 max-w-[323px] hover:scale-[1.01] governance-projects__project"
          href="#"
        >
          <div
            class="card__image-container max-h-[223px] w-full mb-3"
          >
            <div
              class="governance-projects__image-container relative"
            >
              <img
                alt="Contextual Constitutional AI"
                class="governance-projects__image w-full max-h-full object-cover rounded-2xl"
                src="/images/project-winners/akshat-naik.png"
              />
              <div
                class="governance-projects__badge absolute flex items-center gap-1 text-sm font-[650] top-2 left-2 text-bluedot-black bg-cream-normal rounded-lg px-3 py-2"
              >
                <img
                  alt="★"
                  class="governance-projects__star-icon size-[12px]"
                  src="/icons/star_gold.svg"
                />
                 
                1st Place
              </div>
            </div>
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h4
                class="card__title mb-1"
              >
                Contextual Constitutional AI
              </h4>
              <p
                class="card__subtitle text-base"
              >
                by Akshat Naik
              </p>
            </div>
          </div>
        </a>
        <a
          class="card flex flex-col items-start transition-transform duration-200 max-w-[323px] hover:scale-[1.01] governance-projects__project"
          href="#"
        >
          <div
            class="card__image-container max-h-[223px] w-full mb-3"
          >
            <div
              class="governance-projects__image-container relative"
            >
              <img
                alt="Safety Haven: Justifying and exploring an antitrust safe haven for AI safety research collaboration"
                class="governance-projects__image w-full max-h-full object-cover rounded-2xl"
                src="/images/project-winners/ella-duus.png"
              />
              <div
                class="governance-projects__badge absolute flex items-center gap-1 text-sm font-[650] top-2 left-2 text-bluedot-black bg-cream-normal rounded-lg px-3 py-2"
              >
                <img
                  alt="★"
                  class="governance-projects__star-icon size-[12px]"
                  src="/icons/star_gold.svg"
                />
                 
                1st Place
              </div>
            </div>
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h4
                class="card__title mb-1"
              >
                Safety Haven: Justifying and exploring an antitrust safe haven for AI safety research collaboration
              </h4>
              <p
                class="card__subtitle text-base"
              >
                by Ella Duus
              </p>
            </div>
          </div>
        </a>
        <a
          class="card flex flex-col items-start transition-transform duration-200 max-w-[323px] hover:scale-[1.01] governance-projects__project"
          href="#"
        >
          <div
            class="card__image-container max-h-[223px] w-full mb-3"
          >
            <div
              class="governance-projects__image-container relative"
            >
              <img
                alt="Societal Adaptation to AI Human-Labor Automation"
                class="governance-projects__image w-full max-h-full object-cover rounded-2xl"
                src="/images/project-winners/yuval-rymon.png"
              />
              <div
                class="governance-projects__badge absolute flex items-center gap-1 text-sm font-[650] top-2 left-2 text-bluedot-black bg-cream-normal rounded-lg px-3 py-2"
              >
                <img
                  alt="★"
                  class="governance-projects__star-icon size-[12px]"
                  src="/icons/star_gold.svg"
                />
                 
                1st Place
              </div>
            </div>
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h4
                class="card__title mb-1"
              >
                Societal Adaptation to AI Human-Labor Automation
              </h4>
              <p
                class="card__subtitle text-base"
              >
                by Yuval Rymon
              </p>
            </div>
          </div>
        </a>
        <a
          class="card flex flex-col items-start transition-transform duration-200 max-w-[323px] hover:scale-[1.01] governance-projects__project"
          href="#"
        >
          <div
            class="card__image-container max-h-[223px] w-full mb-3"
          >
            <div
              class="governance-projects__image-container relative"
            >
              <img
                alt="Are you secretly training AI? Methods for uncovering covert AI training: A framework for feasibility and future research"
                class="governance-projects__image w-full max-h-full object-cover rounded-2xl"
                src="/images/project-winners/naci-cankaya.png"
              />
              <div
                class="governance-projects__badge absolute flex items-center gap-1 text-sm font-[650] top-2 left-2 text-bluedot-black bg-cream-normal rounded-lg px-3 py-2"
              >
                <img
                  alt="★"
                  class="governance-projects__star-icon size-[12px]"
                  src="/icons/star_gold.svg"
                />
                 
                1st Place
              </div>
            </div>
          </div>
          <div
            class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
          >
            <div
              class="card__text"
            >
              <h4
                class="card__title mb-1"
              >
                Are you secretly training AI? Methods for uncovering covert AI training: A framework for feasibility and future research
              </h4>
              <p
                class="card__subtitle text-base"
              >
                by Naci Cankaya
              </p>
            </div>
          </div>
        </a>
      </div>
      <a
        class="cta-button flex items-center justify-center rounded transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter governance-projects__link h-fit py-3"
        data-testid="cta-link"
        href="#"
      >
        <span
          class="cta-button__text"
        >
          Explore more projects
        </span>
        <span
          class="cta-button__chevron ml-3"
        >
          <img
            alt="→"
            class="cta-button__chevron-icon size-2"
            src="/icons/chevron_blue.svg"
          />
        </span>
      </a>
    </div>
  </section>
</div>
`;
